lib/shopify_app/shop.rb

```diff
@@ -4,12 +4,20 @@ module Shop
 
     included do
       validates :shopify_domain, presence: true, uniqueness: true
-      validates :shopify_token, presence: true
+      if shopify_token_needs_to_exist?
+        validates :shopify_token, presence: true
+      end
     end
 
     def with_shopify_session(&block)
       ShopifyAPI::Session.temp(shopify_domain, shopify_token, &block)
     end
 
+    module ClassMethods
+      def shopify_token_needs_to_exist?
+        true
+      end
+    end
```

Comment:
This code looks fine but my question for you guys is this: How common is this use case and it is even possible for non-internal apps? If this isn't useful to the app community at large then I would prefer to not accept this PR and for you to just have your own shop model concern in your app.